tests/produce_fetch.rs

```diff
@@ -0,0 +1,249 @@
+use std::{net::TcpStream, time::Duration};
+
+use bytes::{Bytes, BytesMut};
+use kafka_protocol::{
+    messages::{
+        create_topics_request::CreatableTopic,
+        fetch_request::{FetchPartition, FetchTopic},
+        produce_request::{PartitionProduceData, TopicProduceData},
+        ApiKey, CreateTopicsRequest, CreateTopicsResponse, FetchRequest, FetchResponse,
+        ProduceRequest, ProduceResponse, RequestHeader, TopicName,
+    },
+    protocol::{Builder, StrBytes},
+    records::{
+        Compression, Record, RecordBatchDecoder, RecordBatchEncoder, RecordEncodeOptions,
+        TimestampType,
+    },
+};
+
+mod common;
+use common::*;
+
+#[test]
+fn record_batch_produce_fetch() {
+    let topic_name = TopicName(StrBytes::from_static_str("record_batch_produce_fetch"));
+
+    let container = start_kafka();
+    let mut socket = connect_to_kafka(&container);
+
+    let records = vec![
+        new_record(0, true),
+        new_record(1, true),
+        new_record(2, true),
+    ];
+
+    let mut encoded = BytesMut::new();
+    RecordBatchEncoder::encode(
+        &mut encoded,
+        records.iter(),
+        &RecordEncodeOptions {
+            version: 2,
+            compression: Compression::None,
+        },
+    )
+    .unwrap();
+
+    create_topic(topic_name.clone(), &mut socket);
+
+    // Sometimes (rarely) the produce request will fail with error code 6 (NOT_LEADER_OR_FOLLOWER)
+    // Hopefully a 1s sleep will prevent that
+    std::thread::sleep(Duration::from_secs(1));
```

Comment:
if this ends up being flaky, we could probably do a topic list in a loop to wait for the node to elect itself?